### PR TITLE
Add testing for PA optional inputs

### PR DESCRIPTION
**Pull request by matthewkotila**

### CODE REVIEW

File: qa/common/perf_analyzer_input_data_json/int_data_optional.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "data" :
+        [
+        
+            {
+                "INPUT0" : [1]
+            }
```
Comment (Tabrizian) [inline qa/common/perf_analyzer_input_data_json/int_data_optional.json:7]: Can we add another line for `INPUT1` too?

```
{
  "INPUT1" : [1]
}
```

Comment (matthewkotila) [reply qa/common/perf_analyzer_input_data_json/int_data_optional.json:7]: It's supposed to be missing to show that PA doesn't error when missing an optional input.

Comment (Tabrizian) [reply qa/common/perf_analyzer_input_data_json/int_data_optional.json:7]:
```
{
    "data" :
        [

            {
                "INPUT0" : [1]
            },
            {
                "INPUT1" : [1]
            }
       ]
}       
```
I mean something like above. Just to make sure if the optional input changes, everything still works properly.

Comment (matthewkotila) [reply qa/common/perf_analyzer_input_data_json/int_data_optional.json]: Added 👍

File: qa/L0_perf_analyzer/test.sh [hunk 2]
```diff
@@ -766,6 +772,22 @@ if [ $(cat $CLIENT_LOG |  grep "${ERROR_STRING}" | wc -l) -ne 0 ]; then
 fi
 set -e
 
+# Test with output validation
+set +e
+$PERF_ANALYZER -v -m optional --measurement-mode "count_windows" \
```
[the remark this answers is not in the document]
Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh:777]: Added this case 👍

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**